Lib/pdb.py

```diff
@@ -306,6 +306,12 @@ class Pdb(bdb.Bdb, cmd.Cmd):
 
     _file_mtime_table = {}
 
+    _last_pdb_instance = None
+
+    def __new__(self, *args, **kwargs):
+        Pdb._last_pdb_instance = super().__new__(self)
+        return Pdb._last_pdb_instance
```

Comment:
what if we're creating this instance while debugging? It would overwrite the instance from the hard coded breakpoint, and this one will be reused at the next hard coded breakpoint. This is not what's intended right?

Reply:
Maybe Pdb._last_pdb_instance should be set in set_trace().

Probably need a test for this scenario too.

Reply:
You mean in `Pdb.set_trace()`? That would make some sense, only enlist the instance when it explicitly takes over the tracing.

However, if the user somehow creates and uses a pdb instance before a `breakpoint()`, we really can't determine the purpose. It could be nested debugging (`debug` command in pdb which I think we should patch to make sure the `_last_pdb_instance` is not modified). Or the user could want a new instance, for example, in the test cases.

Reply:
Not sure I understand. If you set `_last_pdb_instance` in `Pdb.set_trace()` then you get clear semantics, and no other breakpoint interrupts with this. Is that not a good thing?

Reply:
No I agree that setting `_last_pdb_instance ` in `Pdb.set_trace()` is a good approach. And I just realized the nested debugging (`debug` command in pdb) uses `sys.call_tracing` so it probably won't affect anything if we do it in `Pdb.set_trace()`. I'll do the change and see how it works out.

Reply:
And this change actually made the test change unnecessary so it's good. Do you think we need to test more scenarios? I think the case where users explicitly create a `Pdb` instance in the pdb prompt is really rare - what's the point?